portal_tables/sync_datasets.py

```diff
@@ -16,15 +16,35 @@ def add_missing_info(
         "".join(["[", d_id, "](", url, ")"]) if url else ""
         for d_id, url in zip(datasets["DatasetAlias"], datasets["DatasetUrl"])
     ]
+    url_pattern = re.compile(".*(synapse\.org).*")
+    alias_pattern = re.compile("^syn\d{7,8}$")
     datasets["grantName"] = ""
     datasets["themes"] = ""
     datasets["consortia"] = ""
     datasets["pub"] = ""
+    datasets["synapseLink"] = ""
     for _, row in datasets.iterrows():
+        is_in_synapse = None
+        for s in row["link"].split("]("):
+            s_match = re.match(url_pattern, s)
+            if s_match:
+                is_in_synapse = True
+        if is_in_synapse:
+            datasets.at[_, "synapseLink"] = datasets.at[_, "link"]
+        else:
```

Comment:
I think you could simplify this a bit, by using `any()`:

```suggestion
        if any(url_pattern.match(s) for s in row["link"].split("](")):
            datasets.at[_, "synapseLink"] = datasets.at[_, "link"]
        else:
```

Or maybe even simpler with this:

```suggestion
        if "synapse.org" in row["link"]:
            datasets.at[_, "synapseLink"] = datasets.at[_, "link"]
        else:
```

Unless there are downstream implications that require using regex instead?

Reply:
> Unless there are downstream implications that require using regex instead?

This is a good question. I wanted to make sure we had flexibility to identify any URL pattern of interest, in case we want to do a similar extraction for other databases, but I think that can be achieved using your suggestion.

Reply:
Oh, that makes sense! I like it.

I was actually thinking that since these same set of lines are used across multiple scripts, that we could extract them out into the `utils` module as a function.  That way, we would only need to update the code in one place as opposed to multiple sync_* scripts.

Reply:
That sounds great, I've thought about that several times (and I know you've done that for some functions previously) so I'm glad you brought it up!